src/app/common-components/common-components.module.ts

```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { HeaderComponent } from './header/header.component';
+import { FooterComponent } from './footer/footer.component';
+import { NavbarComponent } from './navbar/navbar.component';
+
+@NgModule({
+  declarations: [HeaderComponent, FooterComponent, NavbarComponent],
+  imports: [CommonModule],
+  exports: [HeaderComponent, FooterComponent, NavbarComponent]
```

Comment:
Talking about app-dashboard-layout component, it can have it's module with header and footer declarations.
And you can just import the module into app.module